### PR TITLE
Following PR#925

### PR DESCRIPTION
**Pull request by tadam50**
Signed-off-by: Thomas ADAM <tadam@silicom.fr>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
Using Contingency constructors

**What is the new behavior (if this is a feature change)?**
Using Contingency builder & static methods

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: contingency/contingency-api/src/test/java/com/powsybl/contingency/GeneratorContingencyTest.java [hunk 1]
```diff
@@ -19,16 +23,31 @@
 
     @Test
     public void test() {
-        GeneratorContingency contingency = new GeneratorContingency("id");
+        Contingency contingency = Contingency.shuntCompensator("id");
         assertEquals("id", contingency.getId());
-        assertEquals(ContingencyElementType.GENERATOR, contingency.getType());
+        assertEquals(1, contingency.getElements().size());
```
Comment (mathbagu) [inline contingency/contingency-api/src/test/java/com/powsybl/contingency/GeneratorContingencyTest.java:28]: This is not equivalent. Is it what you wanted to do? It is strange to find a ShuntContingency in GeneratorContingencyTest

File: contingency/contingency-api/src/test/java/com/powsybl/contingency/SubContingenciesProviderTest.java [hunk 2]
```diff
@@ -47,15 +45,30 @@ public void test() {
...
+                        .mapToObj(i -> Contingency.builder("contingency-" + i).build())
+                        .collect(Collectors.toList());
+            }
+
+            @Override
+            public String asScript() {
+                return "";
+            }
+        };
+    }
```
Comment (mathbagu) [inline contingency/contingency-api/src/test/java/com/powsybl/contingency/SubContingenciesProviderTest.java:73]: I don't understand why you change this test. None of these changes have been introduced in #925

File: security-analysis-api/src/test/java/com/powsybl/security/SecurityAnalysisResultBuilderTest.java [hunk 3]
```diff
@@ -59,7 +59,7 @@ public void completeResultWithCustomContext() {
 
         CustomContext postResultContext = new CustomContext(network, "post");
         CustomContext postViolationContext = new CustomContext(network, "post-vio");
-        builder.contingency(new Contingency("contingency1"), postResultContext)
+        builder.contingency(Contingency.builder("contingency1").build(), postResultContext)
```
Comment (mathbagu) [inline security-analysis-api/src/test/java/com/powsybl/security/SecurityAnalysisResultBuilderTest.java:62]: I wonder if for empty contingency we could leave with `new Contingency`

File: contingency/contingency-api/src/test/java/com/powsybl/contingency/GeneratorContingencyTest.java (hunk 1 above)
Comment (tadam50) [reply contingency/contingency-api/src/test/java/com/powsybl/contingency/GeneratorContingencyTest.java:28]: My mistake

File: contingency/contingency-api/src/test/java/com/powsybl/contingency/SubContingenciesProviderTest.java (hunk 2 above)
Comment (tadam50) [reply contingency/contingency-api/src/test/java/com/powsybl/contingency/SubContingenciesProviderTest.java:73]: Bad commit

**Review by mathbagu — APPROVED** [review]

**Pull request merged**